cmake/CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ install(TARGETS ${PROJECT_NAME}
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file("${PROJECT_NAME}ConfigVersion.cmake"
                                  VERSION ${PROJECT_VERSION}
-                                 COMPATIBILITY SameMinorVersion)
+                                 COMPATIBILITY AnyNewerVersion)
```

Comment:
This change doesn't look right. We don't guarantee backward compatibility for all future versions.

Reply:
Same minor doesn’t seem right either. The point is that  when finding a package should request either the exact version you need or the range of versions you support, which might span both minor or major versions

Reply:
@garth-wells  from https://iamsorush.com/posts/cpp-cmake-config/
AnyNewerVersion means CMake imports the package if its version is newer or the same version as the requested one. For more details, see [CMake Manual](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file).

Which means it is up the package requesting UFCx, Basix or DOLFINx to request compatible version ranges

Reply:
We have decided that it is up to the developer of the external library to pin compatible version ranges.